packages/server/src/integrations/queries/sql.ts

```diff
@@ -1,10 +1,30 @@
 import { findHBSBlocks } from "@budibase/string-templates"
-import { DatasourcePlus } from "@budibase/types"
+import { DatasourcePlus, SourceName } from "@budibase/types"
 import sdk from "../../sdk"
 
-const CONST_CHAR_REGEX = new RegExp("'[^']*'", "g")
+function getConstCharRegex(sourceName: SourceName) {
+  // MySQL clients support ANSI_QUOTES mode off, this is by default
+  // but " and ' count as string literals
+  if (sourceName === SourceName.MYSQL) {
+    return new RegExp(`"[^"]*"|'[^']*'`, "g")
```

Comment:
Could you define these regexes as consts at the global level? Constructing them has a non-zero cost.